data_structures/stacks/postfix_evaluation.py

```diff
@@ -17,52 +24,210 @@
         Result =  59
 """
 
-import operator as op
+# Defining valid unary operator symbols
+UNARY_OP_SYMBOLS = ("-", "+")
+
+# Defining valid binary operator symbols
+BINARY_OP_SYMBOLS = ("-", "+", "*", "^", "/")
+
+
+def parse_token(data: str) -> float | str:
+    """
+    Converts the given data to appropriate number if it is indeed a number, else returns
+    the data as it is with a False flag. This function also serves as a check of whether
+    the input is a number or not.
+
+    Parameters
+    ----------
+    data : str
+        The data which needs to be converted to the appropriate number
+
+    Returns
+    -------
+    bool,  int or float
+        Returns a tuple of (a, b) where 'a' is True if data is indeed a number (integer
+        or numeric) and 'b' is either an integer of a floating point number.
+        If 'a' is False, then b is 'data'
+    """
+    try:
+        return int(data)
+    except ValueError:
+        try:
+            return float(data)
+        except ValueError:
+            if is_operator(data):
+                return data
+            msg = f"{data} is neither a number nor a valid operator"
+            raise ValueError(msg)
```

Comment:
```suggestion
    if is_operator(data):
        return data
    try:
        return int(data)
    except ValueError:
        try:
            return float(data)
        except ValueError:
            msg = f"{data} is neither a number nor a valid operator"
            raise ValueError(msg)
```
We can un-nest the operator check

Reply:
I have removed the int check and now it returns only float if number else a string (if operator).